### PR TITLE
CA: Updating AWS-SDK-Go to 1.26.6

### PR DESCRIPTION
**Pull request by gjtempleton**
This brings in the new EC2 `DescribeInstanceTypes` API.

Also updates the AWS-Manager test for Region fetching due to changes in the SDK in how this is done due to https://github.com/aws/aws-sdk-go/pull/2969

Updating vendor against git@github.com:kubernetes/kubernetes.git:711790af23a231ade04c6c39d6bc5dcf1a996f73 (711790af23a231ade04c6c39d6bc5dcf1a996f73)

Broken out from the changes to the EC2 instance type list enabled by this SDK update to simplify reviewing of the PRs.

Related to #2696

### CODE REVIEW

**Comment by Jeffwan** [comment]
Thanks! Seems this is a prerequisite for the next step. Let's merge this dependency update separately. Looks good to me

/lgtm

**Comment by gjtempleton** [comment]
/hold

I've just had a look again and am seeing more changes than I expected, particularly in vendored files under a `k8s.io` path.

Looks like these reversions are all reversions of changes brought in by #2605 . They seem unrelated to the changes intended by that PR, but could you please confirm @jaypipes just to be certain.

**Comment by jaypipes** [comment]
> /hold
> 
> I've just had a look again and am seeing more changes than I expected, particularly in vendored files under a `k8s.io` path.
> 
> Looks like these reversions are all reversions of changes brought in by #2605 . They seem unrelated to the changes intended by that PR, but could you please confirm @jaypipes just to be certain.

@gjtempleton nope, these changes are all good. This PR is adding the new aws-sdk-go 1.26.6 package to the `cluster-autoscaler/vendor/github.com/aws/aws-sdk-go` directory. This is proper, as the `hack/update-vendor.sh` script runs `go mod vendor` after replacing the packages listed in `go-mod.extra` in the go.mod file generated from upstream k/k. #2605 removes the "sub-vendored" `cluster-autoscaler/cloudprovider/digitalocean/godo` package with `cluster-autoscaler/vendor/github.com/digitalocean/godo`.

In short, we're all good here :)

**Comment by gjtempleton** [comment]
Brilliant, thanks for the confirmation!

/hold cancel

**Comment by gjtempleton** [comment]
Closing as this is made redundant by the merging of #2796 

Will port across the code fix for the failing test mentioned in #2797 to a separate PR
